### PR TITLE
In svcat verify service instance exists on unbind (#1744)

### PR DESCRIPTION
**Pull request by witlessbird**

### CODE REVIEW

**Comment by witlessbird** [comment]
This fixes https://github.com/kubernetes-incubator/service-catalog/issues/1744.

**Comment by witlessbird** [comment]
The output of svcat when attempting to execute "unbind" on a non-existing service instance:

```
[wb@trade-surplus svcat{master}]$ ./svcat unbind missing
Error: unable to get instance 'default.missing' (serviceinstances.servicecatalog.k8s.io "missing" not found)
```

**Comment by carolynvs** [comment]
@witlessbird Currently svcat correctly prints a message that the instance couldn't be found. What is the behavior that you were expecting?

**Comment by witlessbird** [comment]
@carolynvs: I see svcat complaining about an instance missing when I try to deprovision an instance. If I try to remove bindings from a non-existent instance, svcat returns without complaints and with an exit status of 0. This PR fixes this behaviour. My apologies, I should've mentioned https://github.com/kubernetes-incubator/service-catalog/issues/1744 somewhere in the comments.

File: pkg/svcat/service-catalog/shared.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package servicecatalog
```
Comment (carolynvs) [inline pkg/svcat/service-catalog/shared.go]: My preference is to leave instance functions in `instance.go`.

**Comment by witlessbird** [comment]
* moved ```RetrieveInstance``` function back into ```instance.go```
* updated commit message to include issue number

**Comment by n3wscott** [comment]
LGTM

**Pull request merged**